### ISSUE TITLE
Updating TF stack results in error if aws_cloudformation_stack resource didn't have any changes

### ISSUE THREAD

**Issue by endofcake** [hashicorp/terraform#12625]
Applying Terraform changes results in error if there are no changes to Cloudformation stack, but there are some changes to other resources.

### Terraform Version
v0.8.7

### Affected Resource(s)
- aws_cloudformation_stack

### Terraform Configuration Files
There's nothing special about my TF files, except the fact that I'm using Cloudformation for my ASG, and Terraform to manage launch configuration and other resources.

### Expected Behavior
When I change some resource (for example, an ECS task) and apply the changes, Terraform should detect that there were no changes to the Cloudformation stack and not try to update it.

### Actual Behavior
Terrafrom tries to update CFN and fails with
```
aws_ecs_task_definition.cadvisor: Creating...
  arn:                         "" => "<computed>"
  ...
aws_ecs_task_definition.cadvisor: Creation complete
Error applying plan:

1 error(s) occurred:

* aws_cloudformation_stack.autoscaling_group: ValidationError: No updates are to be performed.
        status code: 400, request id: xxx
```

### Steps to Reproduce
1. `terraform apply` when no changes to CFN stack were made

### References
Seems to be the same issue as https://github.com/hashicorp/terraform/issues/9081. The difference is that my output had quotes around it, and removing it altogether doesn't solve the issue.

**Comment by evilmarty** [comment]
I can confirm this is occurring in v0.9.2 as well.

```
Error applying plan:
 
2 error(s) occurred:
 
* aws_cloudformation_stack.buildkite_queue[0]: 1 error(s) occurred:
 
* aws_cloudformation_stack.buildkite_queue.0: ValidationError: No updates are to be performed.
	status code: 400, request id: 5f2339d6-24a4-11e7-8e5e-a3a2c9a85c2f
* aws_cloudformation_stack.buildkite_queue[1]: 1 error(s) occurred:
 
* aws_cloudformation_stack.buildkite_queue.1: ValidationError: No updates are to be performed.
	status code: 400, request id: 5f247247-24a4-11e7-a1ab-ef25f4fc1b07
 
Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

**Comment by endofcake** [comment]
As a hacky workaround, I've added a timestamp tag to my CFN so that it's always updated:
```
        - Key: last_updated
          Value: "${timestamp()}"
          PropagateAtLaunch: false
```

**Comment by julianalves** [comment]
AWS changing the description as valid change.

* aws_cloudformation_stack.stack: ValidationError: No updates are to be performed.
	status code: 400, request id: cff26bdd-3a38-11e7-b6e5-91d134971ef0

**Comment by endofcake** [comment]
Fixed in 0.9.6 by https://github.com/hashicorp/terraform/pull/14463.
